darshan-util/pydarshan/tests/test_cffi_misc.py

```diff
@@ -62,3 +63,147 @@ def test_file_hash_type(log_path):
+                0.0, 100397.60042190552, 11.300841808319092, 0.0,
+                17.940945863723755, 20.436099529266357, 85.47495031356812,
+                0.0, 0.0,
+            ]
+        )
+    ]
+)
+@pytest.mark.parametrize("expected_counter_names", [
+        [
+            'POSIX_OPENS', 'POSIX_FILENOS', 'POSIX_DUPS', 'POSIX_READS',
+            'POSIX_WRITES', 'POSIX_SEEKS', 'POSIX_STATS', 'POSIX_MMAPS',
+            'POSIX_FSYNCS', 'POSIX_FDSYNCS', 'POSIX_RENAME_SOURCES',
+            'POSIX_RENAME_TARGETS', 'POSIX_RENAMED_FROM', 'POSIX_MODE',
+            'POSIX_BYTES_READ', 'POSIX_BYTES_WRITTEN', 'POSIX_MAX_BYTE_READ',
+            'POSIX_MAX_BYTE_WRITTEN', 'POSIX_CONSEC_READS',
+            'POSIX_CONSEC_WRITES', 'POSIX_SEQ_READS', 'POSIX_SEQ_WRITES',
+            'POSIX_RW_SWITCHES', 'POSIX_MEM_NOT_ALIGNED',
+            'POSIX_MEM_ALIGNMENT', 'POSIX_FILE_NOT_ALIGNED',
+            'POSIX_FILE_ALIGNMENT', 'POSIX_MAX_READ_TIME_SIZE',
+            'POSIX_MAX_WRITE_TIME_SIZE', 'POSIX_SIZE_READ_0_100',
+            'POSIX_SIZE_READ_100_1K', 'POSIX_SIZE_READ_1K_10K',
+            'POSIX_SIZE_READ_10K_100K', 'POSIX_SIZE_READ_100K_1M',
+            'POSIX_SIZE_READ_1M_4M', 'POSIX_SIZE_READ_4M_10M',
+            'POSIX_SIZE_READ_10M_100M', 'POSIX_SIZE_READ_100M_1G',
+            'POSIX_SIZE_READ_1G_PLUS', 'POSIX_SIZE_WRITE_0_100',
+            'POSIX_SIZE_WRITE_100_1K', 'POSIX_SIZE_WRITE_1K_10K',
+            'POSIX_SIZE_WRITE_10K_100K', 'POSIX_SIZE_WRITE_100K_1M',
+            'POSIX_SIZE_WRITE_1M_4M', 'POSIX_SIZE_WRITE_4M_10M',
+            'POSIX_SIZE_WRITE_10M_100M', 'POSIX_SIZE_WRITE_100M_1G',
+            'POSIX_SIZE_WRITE_1G_PLUS', 'POSIX_STRIDE1_STRIDE',
+            'POSIX_STRIDE2_STRIDE', 'POSIX_STRIDE3_STRIDE',
+            'POSIX_STRIDE4_STRIDE', 'POSIX_STRIDE1_COUNT',
+            'POSIX_STRIDE2_COUNT', 'POSIX_STRIDE3_COUNT',
+            'POSIX_STRIDE4_COUNT', 'POSIX_ACCESS1_ACCESS',
+            'POSIX_ACCESS2_ACCESS', 'POSIX_ACCESS3_ACCESS',
+            'POSIX_ACCESS4_ACCESS', 'POSIX_ACCESS1_COUNT',
+            'POSIX_ACCESS2_COUNT', 'POSIX_ACCESS3_COUNT',
+            'POSIX_ACCESS4_COUNT', 'POSIX_FASTEST_RANK',
+            'POSIX_FASTEST_RANK_BYTES', 'POSIX_SLOWEST_RANK',
+            'POSIX_SLOWEST_RANK_BYTES',
+        ]
+    ]
+)
+@pytest.mark.parametrize("expected_fcounter_names", [
+        [
+            'POSIX_F_OPEN_START_TIMESTAMP', 'POSIX_F_READ_START_TIMESTAMP',
+            'POSIX_F_WRITE_START_TIMESTAMP', 'POSIX_F_CLOSE_START_TIMESTAMP',
+            'POSIX_F_OPEN_END_TIMESTAMP', 'POSIX_F_READ_END_TIMESTAMP',
+            'POSIX_F_WRITE_END_TIMESTAMP', 'POSIX_F_CLOSE_END_TIMESTAMP',
+            'POSIX_F_READ_TIME', 'POSIX_F_WRITE_TIME', 'POSIX_F_META_TIME',
+            'POSIX_F_MAX_READ_TIME', 'POSIX_F_MAX_WRITE_TIME',
+            'POSIX_F_FASTEST_RANK_TIME', 'POSIX_F_SLOWEST_RANK_TIME',
+            'POSIX_F_VARIANCE_RANK_TIME', 'POSIX_F_VARIANCE_RANK_BYTES',
+        ]
```

Comment:
Listing these expected column names explicitly like this seems problematic for a few reasons:

1) It is extremely verbose when they can be replaced by: `backend.counter_names("POSIX")` and `backend.fcounter_names("POSIX")`, respectively
2) Pulling the values directly out of their encoding into the C shared library object using the `backend..` calls allows the expected values to be maintained in one place instead of two--if there's a change in the C library, then there's a higher maintenance burden on the Python code which is maintaining its own list that needs to be changed as well.

I also don't really see a reason to use them as "parameters," could just be one-line assignments to the expected values near the start of the test function proper.

Reply:
I moved all of the single-entry `pytest.mark.parametrize` data into the testing function itself and changed the counter/fcounter names to use their respective functions. 

I would parametrize these if there were more cases to test. At the moment this really only checks 1 log record which isn't super rigorous, I just haven't thought of a good way to check many individual records without print/pasting their value arrays in, which isn't very tenable.

Reply:
I'll open an issue about how you could accomplish this without copy/pasting stuff all over the place. Remember that Shane mentioned that we do have the ability to write arbitrary synthetic data to a new log file, so that would probably be a key component.